File: src/main/java/land/leets/domain/user/presentation/dto/UserDetailsResponse.java
package land.leets.domain.user.presentation.dto;

import land.leets.domain.application.type.SubmitStatus;
import land.leets.domain.user.domain.User;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.UUID;

@Getter
@AllArgsConstructor
public class UserDetailsResponse {

    private UUID uid;
    private String sid;
    private String name;
    private String phone;
    private String email;
    private String major;
    private SubmitStatus submitStatus;

    public static UserDetailsResponse of(User user, SubmitStatus submitStatus) {
        return new UserDetailsResponse(
                user.getUid(),
                user.getSid(),
                user.getName(),
                user.getPhone(),
                user.getEmail(),
                null,
                submitStatus
        );
    }
}

File: src/main/kotlin/land/leets/domain/user/presentation/dto/UserDetailsResponse.kt
package land.leets.domain.user.presentation.dto

import land.leets.domain.application.type.SubmitStatus
import land.leets.domain.user.domain.User
import java.util.UUID

data class UserDetailsResponse(
    val uid: UUID?,
    val sid: String?,
    val name: String,
    val phone: String?,
    val email: String,
    val major: String?,
    val submitStatus: SubmitStatus
) {
    companion object {
        fun of(user: User, submitStatus: SubmitStatus): UserDetailsResponse {
            return UserDetailsResponse(
                uid = user.id,
                sid = user.sid,
                name = user.name,
                phone = user.phone,
                email = user.email,
                major = null,
                submitStatus = submitStatus
            )
        }
    }
}
